examples/humble-turtlebot3-gazebo.nix

```diff
@@ -0,0 +1,23 @@
+# Run:
+# ros2 launch turtlebot3_gazebo turtlebot3_world.launch.py
+# ros2 run turtlebot3_teleop teleop_keyboard
+
+{ pkgs ? import ../. {} }:
+with pkgs;
+with rosPackages.humble;
+with pythonPackages;
+
+mkShell {
+  buildInputs = [
+    glibcLocales
+    (buildEnv { paths = [
+      ros-base
+      turtlebot3-teleop
+      turtlebot3-gazebo
```

Comment:
This package depends on gazebo_ros_pkgs, which is [deprecated](https://index.ros.org/p/gazebo_ros_pkgs/#humble). They recommend to follow https://gazebosim.org/docs/latest/gazebo_classic_migration. Seems like https://gazebosim.org/docs/latest/migrating_gazebo_classic_ros2_packages/ might be what you need.

I also wonder whether #591 contains what you intend to do here or not. There is a new example [ros2-turtlebot4-gz.nix](https://github.com/lopsided98/nix-ros-overlay/pull/591/files#diff-c76790b502439e005313c9d13f55084aa07afec03b708313950637f334fadf91).

Reply:
The proposed pull request certainly tackles some of the issues, and I appreciate that effort! However, I believe it's really important for the layer to create a welcoming experience for new users. If there's a common issue with building widely used packages, it might not provide the best first-time experience for them. I encourage you to take a closer look at my earlier comments and consider the options I've suggested- they're intended to help us all move forward together!